cmd/podman/common/specgen.go

```diff
@@ -439,25 +435,7 @@ func FillOutSpecGen(s *specgen.SpecGenerator, c *ContainerCLIOpts, args []string
 		s.ImageVolumeMode = "anonymous"
 	}
 
-	systemd := c.SystemdD == "always"
-	if !systemd && command != nil {
-		x, err := strconv.ParseBool(c.SystemdD)
-		if err != nil {
-			return errors.Wrapf(err, "cannot parse bool %s", c.SystemdD)
-		}
-		if x && (command[0] == "/usr/sbin/init" || command[0] == "/sbin/init" || (filepath.Base(command[0]) == "systemd")) {
-			systemd = true
-		}
-	}
-	if systemd {
-		if s.StopSignal == nil {
-			stopSignal, err = util.ParseSignal("RTMIN+3")
-			if err != nil {
-				return errors.Wrapf(err, "error parsing systemd signal")
-			}
-			s.StopSignal = &stopSignal
-		}
-	}
+	s.Systemd = c.Systemd
```

Comment:
should we leave the default still to true in `createFlags.StringVar` and set `s.Systemd` only when it is changed?

Reply:
The problem is figuring out if the value changed.  It is a lot of hacking for not much value.  Systemd really does not have a default, Always probably should not exist.  We should have had three values
"true", "false", "" that way if the user forces it to TRUE we setup to run systemd, like we do with Always. "" means the code determines if you are running with systemd mode.

But this is not the default and would be a breaking change. So really "true" means nothing now, or the same as "".

Reply:
We could alias `true` to `always` - I think that 99% of people specifying `--systemd` intended that in the first place?

Reply:
Well almost no one probably specifies that command other then to turn it on for debugging purposes. But lets get this fix in and we can move on.